### ISSUE TITLE
MultiCursor bug

### ISSUE THREAD

**Issue by fedbont94** [matplotlib/matplotlib#20231]
### Bug report

**Bug summary**

When plotting multiple subplots, the cursor it is correctly shown only in the one given. However, it moves regardless of which subplot the mouse is on. In my opinion, this should not be the case. 
I was not able to find a solution to this problem myself. 

**Code for reproduction**

```python
import matplotlib.pyplot as plt
from matplotlib.widgets import MultiCursor

fig = plt.figure(figsize=(18, 12), constrained_layout=True)
gs = fig.add_gridspec(8, 10)

# Let us show 4 subplots:
ax0 = fig.add_subplot(gs[:4, :5])
ax1 = fig.add_subplot(gs[4:, :5])
ax2 = fig.add_subplot(gs[:4, 5:])
ax3 = fig.add_subplot(gs[4:, 5:])

# Shows a cursor for 2 subplots but it moves regardless of which subplot the mouse is on.
multi = MultiCursor(fig.canvas, (ax0, ax1), color='r', lw=1, horizOn=False, vertOn=True)

plt.show()
```

**Actual outcome**
Cursor moves regardless of the subplot 

**Expected outcome**
It should move only when the mouse is on the given subplots

**Matplotlib version**

  * Operating system: Ubuntu 20
  * Matplotlib version (3.3.4): 
  * Matplotlib backend (`Qt5Agg`): 
  * Python version: Python 3.8.6
  * Jupyter version (if applicable): None
  * Other libraries: None

 pip
 None

**Comment by ianhi** [comment]
Hi @fedbont94 good catch - happily this has been fixed already! See https://github.com/matplotlib/matplotlib/issues/19635 and https://github.com/matplotlib/matplotlib/pull/19636. Although I'm not sure which version it will be released in?

So I'll close but feel free to re-open if you think this should be kept open.

**Comment by ianhi** [comment]
Update: It will be released in in `3.4.3` see https://github.com/matplotlib/matplotlib/pull/20232

**Comment by fedbont94** [comment]
Thanks for the help guys!
